### PR TITLE
feat: Q-Law guidance law

### PR DESCRIPTION
**Pull request by vishwa2710**
- [x] Add finite difference solver once that MR is landed
- [x] Add unit tests for Q-Law
- [x] Add unit tests for COE
- [x] Add python bindings for Q-Law
- [x] Add unit tests for python bindings

### CODE REVIEW

**Review by phc1990 — COMMENTED** [review]
LGTM

**Comment by Derollez** [comment]
Do we cases of reference to "validate" the output of Qlaw?

File: include/OpenSpaceToolkit/Astrodynamics/Dynamics/Thruster/GuidanceLaw/ConstantThrust.hpp [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+    /// @brief                  Print constant thrust thruster dynamics
+    ///
+    /// @param                  [in] anOutputStream An output stream
+    /// @param                  [in] displayDecorator A boolean indicating whether or not to display the decorator
+    /// during printing
+
+    virtual void print(std::ostream& anOutputStream, bool displayDecorator = true) const override;
+
+    /// @brief                  Intrack Constant thrust dynamics
```
Comment (apaletta3) [inline include/OpenSpaceToolkit/Astrodynamics/Dynamics/Thruster/GuidanceLaw/ConstantThrust.hpp:101]: Explain that true means forward in track and false means backward in track

File: include/OpenSpaceToolkit/Astrodynamics/Dynamics/Thruster/GuidanceLaw.hpp
Comment (apaletta3) [inline include/OpenSpaceToolkit/Astrodynamics/Dynamics/Thruster/GuidanceLaw.hpp:1]: Would this guidance law interface (abstract class) be flexible enough to allow us to implement a non-thrusting guidance law in the future? For example if we wanted to do a differential drag guidance law. I think the answer is yes in general, but maybe the  `aThrustAcceleration` input for `computeAcceleration` would be renamed as a "control" input acceleration for that? Just food for that.

File: include/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Kepler/COE.hpp [hunk 2]
```diff
@@ -186,6 +186,25 @@ class COE
...
+    /// @return                 Radiaul Distance
+
+    Length getRadialDistance() const;
+
+    /// @brief                  Get Angular Momentum
+    ///
+    /// @return                 Angular Momentum
+
+    // TBM: Maybe return with unit
+    Real getAngularMomentum(const Derived& aGravitationalParameter) const;
```
[the remark this answers is not in the document]
Comment (vishwa2710) [reply include/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Kepler/COE.hpp]: So the static functions return values in SI units. The member functions hold on to the units. I've updated the docstring for the Static methods to reflect that.

File: include/OpenSpaceToolkit/Astrodynamics/Dynamics/Thruster/GuidanceLaw.hpp [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    virtual Vector3d computeAcceleration(
+        const Instant& anInstant,
+        const Vector3d& aPositionCoordinates,
+        const Vector3d& aVelocityCoordinates,
+        const Real& aThrustAcceleration,
+        const Shared<const Frame>& outputFrameSPtr
+    ) const = 0;
+
+   protected:
+    const String name_;
```
[the remark this answers is not in the document]
Comment (vishwa2710) [reply include/OpenSpaceToolkit/Astrodynamics/Dynamics/Thruster/GuidanceLaw.hpp]: Yeah and also just for a nice description when pretty printing.

**Review by kyle-cochran — APPROVED** [review]

**Pull request merged**